nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/QueryDatabaseTable.java

```diff
@@ -190,8 +222,11 @@ public void onTrigger(final ProcessContext context, final ProcessSessionFactory
 
         final DBCPService dbcpService = context.getProperty(DBCP_SERVICE).asControllerService(DBCPService.class);
         final DatabaseAdapter dbAdapter = dbAdapters.get(context.getProperty(DB_TYPE).getValue());
-        final String tableName = context.getProperty(TABLE_NAME).evaluateAttributeExpressions().getValue();
+
+        final String propTableName = context.getProperty(TABLE_NAME).evaluateAttributeExpressions().getValue();
+        final String tableName = StringUtils.isEmpty(propTableName) ? ARBITRARY_SQL_TABLE_NAME : propTableName;
         final String columnNames = context.getProperty(COLUMN_NAMES).evaluateAttributeExpressions().getValue();
+        final String sqlQuery = context.getProperty(SQL_QUERY).getValue();
```

Comment:
Should this support Expression Language for Variable Registry support (see my comment above)?

Reply:
Makes sense to me.